.claude/skills/productize-chart-env/scripts/gen-schema.py

```
@@ -171,7 +176,11 @@ def open_map_schema(child_path, node, desc):
         cs["description"] = cd
     leaf = child_path.split(".")[-1]
     if ENV_KEYS and leaf in ("configmap", "data"):
-        allow = {"enum": sorted(ENV_KEYS)}
+        # The enum is env/rendered keys UNIONed with the block's OWN shipped sub-keys — a
+        # chart may carry reserved non-env keys under `configmap` (e.g. `annotations` for the
+        # ConfigMap metadata) that are valid by construction and must not be rejected.
+        own = {k for k in node} if isinstance(node, dict) else set()
```

Ruff (0.16.1)

[warning] Unnecessary set comprehension (rewrite using `set()`)

Rewrite using `set()`

(C416)